torch/nn/attention/_score_mod.py

```diff
@@ -0,0 +1,64 @@
+"""This module defines the common used score_mod functions for templated attention in PyTorch."""
+import torch
+
+
+def identity(
+    score: torch.Tensor,
+    batch: torch.Tensor,
+    head: torch.Tensor,
+    token_q: torch.Tensor,
+    token_kv: torch.Tensor,
+) -> torch.Tensor:
+    return score
+
+
+def causal(
+    score: torch.Tensor,
+    batch: torch.Tensor,
+    head: torch.Tensor,
+    token_q: torch.Tensor,
+    token_kv: torch.Tensor,
+) -> torch.Tensor:
+    return torch.where(token_q >= token_kv, score, float("-inf"))
+
+
+def rel_bias(
+    score: torch.Tensor,
+    batch: torch.Tensor,
+    head: torch.Tensor,
+    token_q: torch.Tensor,
+    token_kv: torch.Tensor,
+) -> torch.Tensor:
+    return score + (token_q - token_kv)
+
+
+def rel_causal(
+    score: torch.Tensor,
+    batch: torch.Tensor,
+    head: torch.Tensor,
+    token_q: torch.Tensor,
+    token_kv: torch.Tensor,
+) -> torch.Tensor:
+    return torch.where(token_q <= token_kv, score + (token_q - token_kv), float("-inf"))
+
+
+def alibi_bias(
+    score: torch.Tensor,
+    batch: torch.Tensor,
+    head: torch.Tensor,
+    token_q: torch.Tensor,
+    token_kv: torch.Tensor,
+) -> torch.Tensor:
+    return score + (token_q - token_kv) * head
```

Comment:
Also this isn't really alib attention yet, https://github.com/drisspg/transformer_nuggets/blob/c11220bbdd14a3f139013b91beec1f8e969b506a/transformer_nuggets/flash/masks.py#L67

this would be actual alibi bias if the num_heads was captured, so not sure if we want to add these just yet

Reply:
Yes, but this function's name is ```_alibi_bias```, is that right?